src/MizarData/DummyFunctionSymbolToKey.h

```diff
@@ -0,0 +1,47 @@
+// Copyright (c) 2022 The Orbit Authors. All rights reserved.
+// Use of this source code is governed by a BSD-style license that can be
+// found in the LICENSE file.
+
+#ifndef MIZAR_DATA_DUMMY_FUNCTION_SYMBOL_TO_KEY_H_
+#define MIZAR_DATA_DUMMY_FUNCTION_SYMBOL_TO_KEY_H_
+
+#include <absl/container/flat_hash_map.h>
+#include <absl/container/flat_hash_set.h>
+
+#include <string>
+
+#include "MizarBase/FunctionSymbols.h"
+
+namespace orbit_mizar_data {
+
+// If a function comes from `mappableModules`, the key from `functionNameToKey` is returned if
+// present. Defaulting to the function name as the key.
+class DummyFunctionSymbolToKey {
+  using FunctionSymbol = ::orbit_mizar_base::FunctionSymbol;
+
+ public:
+  DummyFunctionSymbolToKey(
+      const absl::flat_hash_map<std::string, std::string>* function_name_to_key,
+      const absl::flat_hash_set<std::string>* mappable_modules)
+      : function_name_to_key_(function_name_to_key), mappable_modules_(mappable_modules) {}
+
+  [[nodiscard]] std::string GetKey(const FunctionSymbol& symbol) const;
+
+ private:
+  const absl::flat_hash_map<std::string, std::string>* function_name_to_key_;
+  const absl::flat_hash_set<std::string>* mappable_modules_;
+};
+
+class D3D11DummyFunctionSymbolToKey : public DummyFunctionSymbolToKey {
+ public:
+  D3D11DummyFunctionSymbolToKey()
+      : DummyFunctionSymbolToKey(kDirectXToDxvkNames, kMappableModules) {}
```

Comment:
This looks a bit dangerous: The initialization order for static global variables is undefined. So if you create a D3D11DummyFunctionSymbolToKey at static initilzation time somewhere you cannot be sure that kDirectXToDxvkNames and kMappableModules already exist.

Easy workaround: define private static methods that return a local static pointer:

```
private:
  static absl::flat_hash_map<std::string, std::string>* GetDirectXToDxvkNames();
```

and in the implementation:
```
absl::flat_hash_map<std::string, std::string>* GetDirectXToDxvkNames() {
  static const absl::flat_hash_map<std::string,
                          std::string>* kDirectXToDxvkNames =
    new absl::flat_hash_map<std::string, std::string>{
        {"CContext::TID3D11DeviceContext_ClearRenderTargetView_<1>",
        ...
   return kDirectXToDxvkNames ;
}
```

You can then 
```
D3D11DummyFunctionSymbolToKey()
      : DummyFunctionSymbolToKey(GetDirectXToDxvkNames(), ...) {}
```

Reply:
Didn't think about it, thanks for the great point! I don't have a static instance of `D3D11DummyFunctionSymbolToKey` (neither I plan to), so it might be fine as is. 

@florian-kuebler my initial template-based solution wouldn't suffer from this issue, as it captured by `auto&`. Would you still prefer the version including Daniel's fix over the original one?